ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -0,0 +1,205 @@
+package ray
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	fmtErrors "github.com/pkg/errors"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+)
+
+// RayJobReconciler reconciles a RayJob object
+type RayJobReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+// NewRayJobReconciler returns a new reconcile.Reconciler
+func NewRayJobReconciler(mgr manager.Manager) *RayJobReconciler {
+	return &RayJobReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+		Log:    ctrl.Log.WithName("controllers").WithName("RayJob"),
+	}
+}
+
+// Reconcile reads that state of a RayJob object and makes changes based on it
+// and what is in the RayJob.Spec
+// Automatically generate RBAC rules to allow the Controller to read and write workloads
+// +kubebuilder:rbac:groups=ray.io,resources=rayjobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ray.io,resources=rayjobs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=ray.io,resources=rayjobs/finalizer,verbs=update
+// +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods/status,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=services/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;create;update
+// +kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=get;list;watch;create;delete
+// +kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=roles,verbs=get;list;watch;create;delete;update
+// +kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=rolebindings,verbs=get;list;watch;create;delete
+// Reconcile used to bridge the desired state with the current state
+func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	_ = r.Log.WithValues("rayjob", request.NamespacedName)
+	log.Info("reconciling RayJob", "service NamespacedName", request.NamespacedName)
```

Comment:
is this log from RayCluster?

Reply:
Yes. https://github.com/ray-project/kuberay/blob/68faa10bc4af44c565fda9fab731f0de0935287e/ray-operator/controllers/ray/raycluster_controller.go#L40

It should be changed to Logger of current controller. BTW, I will make some changes to remove package scope `log` in raycluster. It's kind of annoying.

Reply:
Pls fix this and see how RayService uses its own logger